https-proxy-agent.js

```diff
@@ -36,6 +36,9 @@ function HttpsProxyAgent (opts) {
   // if `true`, then connect to the proxy server over TLS. defaults to `false`.
   this.secureProxy = proxy.protocol ? /^https:?$/i.test(proxy.protocol) : false;
 
+  // use custom CA certificate
+  this.ca = opts.ca;
```

Comment:
Shouldn't the `ca` be passed in to the `https.request()` (or fetch() or whatever accepts the agent instance) options object instead of the Agent constructor?